itests/kit/deals.go

```diff
@@ -394,7 +396,7 @@ func (dh *DealHarness) ExtractFileFromCAR(ctx context.Context, file *os.File) st
 	b, err := bserv.GetBlock(ctx, ch.Roots[0])
 	require.NoError(dh.t, err)
 
-	nd, err := ipld.Decode(b)
+	nd, err := dh.ipldDecoder.DecodeNode(ctx, b)
```

Comment:
Using a global registry here seems unintentional/was easy. This mostly swaps which globals you're using (and which codec implementations), but if you knew the set you wanted to support this could also be handled explicitly (e.g. adding support for dag-pb, raw, dag-cbor).